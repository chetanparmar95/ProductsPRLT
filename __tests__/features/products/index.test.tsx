
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ProductList from '../../../src/features/products';

const mockStore = configureStore([]);
const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe('ProductList Component', () => {
  it('ProductList renders correctly', () => {
    const store = mockStore({
        products: {
          products: [
            {
              id: 1,
              name: 'Product 1',
              price: 10.0,
              quantity: 0,
              colour: "Black",
              img: "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024"
            },
            {
              id: 2,
              name: 'Product 2',
              price: 15.0,
              quantity: 0,
              colour: "Black",
              img: "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024"
            },
          ],
          cartItems: []
        },
    });
    const { toJSON, getByText } = render(
        <Provider store={store}>
            <ProductList />
        </Provider>
    );
    expect(toJSON()).toMatchSnapshot();

    const product1Text = getByText('Product 1');
    const product2Text = getByText('Product 2');

    expect(product1Text).toBeDefined();
    expect(product2Text).toBeDefined();
  });

  it('Check if cart button navigate to cart screen', () => {
    const store = mockStore({
      products: {
        products: [
          {
            id: 1,
            name: 'Product 1',
            price: 10.0,
            quantity: 0,
            colour: "Black",
            img: "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024"
          },
          {
            id: 2,
            name: 'Product 2',
            price: 15.0,
            quantity: 0,
            colour: "Black",
            img: "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024"
          },
        ],
        cartItems: []
      },
    });
    const { getByText } = render(
        <Provider store={store}>
            <ProductList />
        </Provider>
    );
    const cartButton = getByText('0 item in Cart');
    fireEvent.press(cartButton)
    expect(mockedNavigate).toHaveBeenCalledTimes(1)
  });
})
