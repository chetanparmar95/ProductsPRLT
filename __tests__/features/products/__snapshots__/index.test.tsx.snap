// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component ProductList renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "margin": 16,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "colour": "Black",
          "id": 1,
          "img": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
          "name": "Product 1",
          "price": 10,
          "quantity": 0,
        },
        {
          "colour": "Black",
          "id": 2,
          "img": "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024",
          "name": "Product 2",
          "price": 15,
          "quantity": 0,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 3,
              "flexDirection": "row",
              "marginVertical": 10,
              "padding": 15,
              "shadowColor": "#333",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 2,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "http://cdn-img.prettylittlething.com/9/0/a/a/90aa90903a135ee59594f47c7685aa7ef3046e44_cly8063_1.jpg?imwidth=1024",
              }
            }
            style={
              {
                "borderRadius": 10,
                "height": 160,
                "width": 100,
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
                "marginLeft": 10,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "width": "100%",
                }
              }
            >
              Product 1
            </Text>
            <Text
              style={
                {
                  "color": "#888",
                  "fontSize": 14,
                  "marginTop": 5,
                }
              }
            >
              Color: 
              Black
            </Text>
            <Text
              style={
                {
                  "color": "#FF5722",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginTop": 5,
                }
              }
            >
              Price: $
              10
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#FF5722",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                    "marginLeft": 5,
                  }
                }
              >
                Add to Cart
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 3,
              "flexDirection": "row",
              "marginVertical": 10,
              "padding": 15,
              "shadowColor": "#333",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 2,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              {
                "uri": "https://cdn-img.prettylittlething.com/d/c/3/3/dc337260f9ecefdb99a8c8e98cd73ccb1b79cea5_cmb6804_4.jpg?imwidth=1024",
              }
            }
            style={
              {
                "borderRadius": 10,
                "height": 160,
                "width": 100,
              }
            }
          />
          <View
            style={
              {
                "flex": 1,
                "marginLeft": 10,
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "width": "100%",
                }
              }
            >
              Product 2
            </Text>
            <Text
              style={
                {
                  "color": "#888",
                  "fontSize": 14,
                  "marginTop": 5,
                }
              }
            >
              Color: 
              Black
            </Text>
            <Text
              style={
                {
                  "color": "#FF5722",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginTop": 5,
                }
              }
            >
              Price: $
              15
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#FF5722",
                  "borderRadius": 5,
                  "flexDirection": "row",
                  "marginTop": 10,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontWeight": "bold",
                    "marginLeft": 5,
                  }
                }
              >
                Add to Cart
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF5722",
        "borderRadius": 20,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 16,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#fff",
            "fontWeight": "bold",
            "marginLeft": 5,
          },
          {},
        ]
      }
    >
      0 item in Cart
    </Text>
  </View>
</View>
`;
